src/submission/submission-service.ts

```diff
@@ -279,51 +293,161 @@ export namespace operations {
     const updatedClinicalEntites: ClinicalEntities = clearClinicalEnitytStats(
       exsistingActiveSubmission.clinicalEntities,
     );
+
     const createdAt: DeepReadonly<Date> = new Date();
-    const schemaErrors: { [k: string]: SubmissionValidationError[] } = {}; // object to store all errors for entity
+
+    // object to store all errors for entity
+    const schemaErrors: { [k: string]: SubmissionValidationError[] } = {};
+
     for (const [clinicalType, newClinicalEnity] of Object.entries(filteredClinicalEntites)) {
-      const { schemaErrorsTemp, processedRecords } = await checkClinicalEntity({
-        records: newClinicalEnity.records,
-        programId: command.programId,
-        clinicalType: clinicalType,
-      });
+      const { schemaErrorsTemp, processedRecords } = await checkClinicalEntity(
+        {
+          records: newClinicalEnity.records,
+          programId: command.programId,
+          clinicalType: clinicalType,
+        },
+        schemaManager.instance().getCurrent(),
+      );
+
+      // because there was a requirement to not keep an open empty submission
+      // we have to return a fake submission object in case there are schema errors
+      // after the update/delete submission is callled below
       if (schemaErrorsTemp.length > 0) {
         // store errors found and remove clinical type from clinical entities
         schemaErrors[clinicalType] = schemaErrorsTemp;
         delete updatedClinicalEntites[clinicalType];
-      } else {
-        // update or add entity
-        updatedClinicalEntites[clinicalType] = {
-          batchName: newClinicalEnity.batchName,
-          creator: newClinicalEnity.creator,
-          createdAt: createdAt,
-          records: processedRecords,
-          ...emptyStats,
-        };
+        continue;
       }
+
+      // update or add entity
+      updatedClinicalEntites[clinicalType] = {
+        batchName: newClinicalEnity.batchName,
+        creator: newClinicalEnity.creator,
+        createdAt: createdAt,
+        schemaErrors: [],
+        records: processedRecords as any,
+        ...emptyStats,
+      };
     }
+
     const newActiveSubmission: ActiveClinicalSubmission = {
       programId: command.programId,
       state: SUBMISSION_STATE.OPEN,
       version: '', // version is irrelevant here, repo will set it
       clinicalEntities: updatedClinicalEntites,
       updatedBy: command.updater,
     };
+
     // insert into database
-    const updated = await submissionRepository.updateSubmissionWithVersion(
+    let updated = (await updateSubmissionWithVersionOrDeleteEmpty(
       command.programId,
       exsistingActiveSubmission.version,
       newActiveSubmission,
-    );
+    )) as ActiveClinicalSubmission;
+
+    if (updated == undefined) {
```

Comment:
This one needs to have `===`, pretty sure lots of things can match to undefined on a `==`